src/hostcalls.rs

```diff
@@ -718,7 +756,7 @@ pub fn send_http_response(
             body.map_or(0, |body| body.len()),
             serialized_headers.as_ptr(),
             serialized_headers.len(),
-            -1,
+            http_to_grpc_status_code(status_code),
```

Comment:
I don't believe this is correct. It appear to be always sending gRPC status code, even for non-gRPC responses.

Reply:
that's true, is it possible to know if it is grpc in this point? or in the wasm calling this? If I create a wasm that just create a local reply and I make a call with grpcurl, the error I get is the code -1, then current implementation is always sending an status code and envoy is forwarding to the client. How do you think we should fix this scenario?

Reply:
Interesting, there is already a mapping in envoy (not the same I have created) 
https://github.com/envoyproxy/envoy/blob/main/source/common/grpc/status.cc#L6

and I guess from envoy doc,  will use this value to override that
https://github.com/envoyproxy/envoy/blob/main/envoy/http/filter.h#L579

What seams to be happening is that in the wasm context the value sent from this function (-1). is directly sent here
https://github.com/envoyproxy/envoy/blob/main/source/extensions/common/wasm/context.cc#L1672

and set always a value to the optional, and does not let envoy to automatically map to the corresponding grpc code from http code, because any value sent will override that

Reply:
probably it would be cleaner if we can instruct envoy to interpret -1 as null (because 0 is a valid status code) to forward that an an empty optional and let envoy do it's magic., saying that, I still think that sending a meaning value could be better than sending -1.

Reply:
Let me shared the test I have done, first a call without the change, and after that calls with a wasm filter that do just basic auth, the first request show the -1 (4294967295) currently sent by the wasm without the change

Old wasm

jolle@jolle-ltmrmbv InstallFromScratch % grpcurl -d '{"greeting": "pepe"}' --plaintext -protoset grpc/hello.proto-descriptor_grpc localhost:8080  hello.HelloService.SayHello
ERROR:
  Code: Internal
  Message: transport: malformed grpc-status: strconv.ParseInt: parsing "4294967295": value out of range

New Wasm rejecting the call no auth header

% grpcurl -d '{"greeting": "pepe"}' --plaintext -protoset grpc/hello.proto-descriptor_grpc localhost:8080  hello.HelloService.SayHello
ERROR:
  Code: Unauthenticated
  Message: {"error":"Registered authentication is set to HTTP basic authentication but there was no security context on the session."}

New wasm with invalid user:pass

% grpcurl -expand-headers -H 'Authorization: Basic YXNkOmFzZA==' -d '{"greeting": "mulesoft"}' --plaintext -protoset grpc/hello.proto-descriptor_grpc localhost:8080  hello.HelloService.SayHello
ERROR:
  Code: Unauthenticated
  Message: {"error":"Authentication Attempt Failed"}


Server rejected the query (proto not corresponding with server)

% grpcurl -expand-headers -H 'Authorization: Basic YWRtaW46dGVzdA==' -d '{"greeting": "pepe"}' --plaintext -protoset grpc/hello.proto-descriptor_grpc localhost:8080  hello.HelloService.SayHello
ERROR:
  Code: Unimplemented
  Message: unknown service hello.HelloService

OK

% grpcurl -expand-headers -H 'Authorization: Basic YWRtaW46dGVzdA==' -d '{"greeting": "pepe"}' --plaintext -protoset grpc/hello.proto-descriptor_grpc localhost:8080  hello.HelloService.SayHello
{
  "reply": "hello pepe"
}